services/order.py

```diff
@@ -0,0 +1,32 @@
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import User, Order, Ticket
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None
+) -> Order:
+    with transaction.atomic():
```

Comment:
In case you would like the whole body of the function to be treated as an atomic transaction, you could use `transaction.atomic` as a decorator.